src/Symfony/Component/Console/Event/ConsoleSignalEvent.php

```diff
@@ -21,15 +21,34 @@
 final class ConsoleSignalEvent extends ConsoleEvent
 {
     private int $handlingSignal;
+    private ?int $exitCode;
 
-    public function __construct(Command $command, InputInterface $input, OutputInterface $output, int $handlingSignal)
+    public function __construct(Command $command, InputInterface $input, OutputInterface $output, int $handlingSignal, ?int $exitCode = 0)
     {
         parent::__construct($command, $input, $output);
         $this->handlingSignal = $handlingSignal;
+        $this->exitCode = $exitCode;
     }
 
     public function getHandlingSignal(): int
     {
         return $this->handlingSignal;
     }
+
+    /**
+     * Sets whether to automatically exit with $exitCode after command handling, or not with null.
```

Comment:
The main feature is to exit, the code is secondary.
```suggestion
     * Sets whether to automatically exit after command handling with $exitCode, or not exit with null.
```

The method could be renamed to `function exit(int|bool $exitCode = true)`:

```php
// No exit
$event->exit(false);

// Exit with success (code=0)
$event->exit();

// Exit with custom code
$event->exit(1);
```

Reply:
This may lead to some poor implementations (`$event->exit(1)` for example) and typecasting bugs. IMO multitype arguments is evil. I suggest logic more like `stopPropagation` on standart events.

We can call something like `$event->setExitCode(int)` / `$event->abortExit()` and leave developer to decide whenether to exit and with wich code

Reply:
This API could work with the event, but not with the SignalableCommandInterface. I prefer to have the same API on both way to deal with signals.

Reply:
But having two method is OK 👍🏼